File: tools/resolve_conflicts.py
#!/usr/bin/env python3
import re
import sys
from pathlib import Path

root = Path(__file__).resolve().parent.parent

conflict_marker = re.compile(r"^$", re.M)
end_re = re.compile(r"^ .*$(\n)?", re.M)

fixed_files = []

for p in root.rglob('*'):
    if not p.is_file():
        continue
    try:
        text = p.read_text(encoding='utf-8')
    except Exception:
        continue
    if '<<<<<<<' not in text:
        continue
    original = text
    changed = True
    while '<<<<<<<' in text and changed:
        changed = False
        m_start = start_re.search(text)
        if not m_start:
            break
        m_mid = mid_re.search(text, m_start.end())
        if not m_mid:
            break
        m_end = end_re.search(text, m_mid.end())
        if not m_end:
            break
        # extract theirs block (between mid and end)
        theirs = text[m_mid.end():m_end.start()]
        # replace whole conflict with theirs
        text = text[:m_start.start()] + theirs + text[m_end.end():]
        changed = True
    if text != original:
        # backup
        try:
            bkp = p.with_suffix(p.suffix + '.orig')
            if not bkp.exists():
                p.rename(bkp)
                bkp.write_text(original, encoding='utf-8')
                # write fixed content to original path
                bkp.replace(p)
            else:
                p.write_text(text, encoding='utf-8')
        except Exception:
            p.write_text(text, encoding='utf-8')
        fixed_files.append(str(p.relative_to(root)))

if fixed_files:
    print('Fixed files:')
    for f in fixed_files:
        print(f)
    sys.exit(0)
else:
    print('No conflict markers found.')
    sys.exit(0)

File: tools/resolve_conflicts.js
const fs = require('fs');
const path = require('path');

const root = path.resolve(__dirname, '..');

function walk(dir) {
  let results = [];
  const list = fs.readdirSync(dir, { withFileTypes: true });
  for (const dirent of list) {
    const full = path.join(dir, dirent.name);
    if (dirent.isDirectory()) {
      results = results.concat(walk(full));
    } else {
      results.push(full);
    }
  }
  return results;
}

const files = walk(root);
const fixed = [];
for (const f of files) {
  let text;
  try { text = fs.readFileSync(f, 'utf8'); } catch (e) { continue; }
  if (!text.includes('', start);
    const end = text.indexOf('', mid);
    if (start === -1 || mid === -1 || end === -1) break;
    const theirs = text.substring(mid + '======='.length, end);
    text = text.substring(0, start) + theirs + text.substring(end + ('>>>>>>>'.length));
    changed = true;
  }
  if (text !== orig) {
    try {
      fs.copyFileSync(f, f + '.orig');
    } catch (e) {}
    fs.writeFileSync(f, text, 'utf8');
    fixed.push(path.relative(root, f));
  }
}

if (fixed.length) {
  console.log('Fixed files:');
  fixed.forEach(f => console.log(f));
} else {
  console.log('No conflict markers found.');
}
